File: Module 1 Foundations/Patterns/Patterns.py
# Pattern 1
for i in range(5):
    for j in range(5):
        print("*", end="") #for printing without line breaks
    print("")

# Pattern 2
for i in range(5):
    for j in range(i+1):
        print("*", end="")
    print("")

# Pattern 3
for i in range(1,6):
    for j in range(1,i+1):
        print(j, end="")
    print("")

# Pattern 4
for i in range(1,6):
    for j in range(1,i+1):
        print(i, end="")
    print("")

# Pattern 5
for i in range(5,0, -1):
    for j in range(1,i+1):
        print("*", end="")
    print("")

# Pattern 6
for i in range(5,0, -1):
    for j in range(1,i+1):
        print(j, end="")
    print("")

File: Module 1 Foundations/Patterns/Patterns.js
// *****
// *****
// *****
// *****
// *****

for (let i = 0; i < 5; i++) {
  for (let j = 0; j < 5; j++) {
    process.stdout.write("*"); // print without line break
  }
  console.log("");
}

// *
// **
// ***
// ****
// *****

for (let i = 0; i < 5; i++) {
  for (let j = 0; j <= i; j++) {
    process.stdout.write("*");
  }
  console.log("");
}

// 1
// 12
// 123
// 1234
// 12345

for (let i = 1; i <= 5; i++) {
  for (let j = 1; j <= i; j++) {
    process.stdout.write(`${j}`); //Simply using j inside write throws errors
  }
  console.log("");
}

// 1
// 22
// 333
// 4444
// 55555

for (let i = 1; i <= 5; i++) {
  for (let j = 1; j <= i; j++) {
    process.stdout.write(`${i}`);
  }
  console.log("");
}

// *****
// ****
// ***
// **
// *
for (let i = 5; i > 0; i--) {
  for (let j = 0; j < i; j++) {
    process.stdout.write("*");
  }
  console.log("");
}

// 12345
// 1234
// 123
// 12
// 1

for (let i = 5; i > 0; i--) {
  for (let j = 1; j <= i; j++) {
    process.stdout.write(`${j}`);
  }
  console.log("");
}
